pkg/apis/pipeline/v1alpha1/pull_request_resource.go

```diff
@@ -100,8 +100,10 @@ func (s *PullRequestResource) GetInputTaskModifier(ts *TaskSpec, sourcePath stri
 }
 
 func (s *PullRequestResource) GetOutputTaskModifier(ts *TaskSpec, sourcePath string) (TaskModifier, error) {
+	// When used as an output, the PR resource should initially fill the output directory with the PR contents.
 	return &InternalTaskModifier{
-		StepsToAppend: s.getSteps("upload", sourcePath),
```

Comment:
Unless I missed something, I think this makes it impossible to use this resource as output, as the output resource steps are executed last, so after all the work in the Task is complete, the resource will now download the PR to the output folder, overwriting any previous work there, and then upload it. 

We probably need some integration test as well, since this should have been detected by testing.

Reply:
I'm not sure I see the issue (although we do need integration testing).

I *think* this means that before the task runs, we "prepend" the "download" step. So we fill that directory with contents. Then after the task runs, we "append" the "upload" step, which uploads those directory contents.

Does that make sense, or am I still missing something?

Reply:
@dlorenc sorry, my bad. I didn't realize that resources had the ability to prepend steps now. So this looks good. I'm happy for integration testing to be added as a follow-up.